### PR TITLE
Day19-HoaVo-MediaPlayer

### PR DESCRIPTION
**Pull request by at-hoavo**
#### Related issues (Optional)
Task #100

#### Description (Required)
 - [x] Play a list of songs ( online ),
 - [x] Have function: `Next` - `Previous` - `Play` - `Shuffle` - `Auto Next`
 - [x]  When press `Lockscreen` -> Show notification with:
         - Image's song, name's song, name's singer
         - Buton `close notification`, and when press it -> stop `play music`
         - Song's progress ( show song's duration and current time )

### CODE REVIEW

**Comment by at-hoavo** [comment]
@at-vinhhuynh please check for me,thank you!

File: app/src/main/java/vn/asiantech/internship/day19/adapter/ChooseSongListener.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package vn.asiantech.internship.day19.adapter;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by rimoka on 03/07/2017.
+ */
+public interface ChooseSongListener {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/adapter/ChooseSongListener.java]: Please change interface name to `OnChooseSongListener`
And method name to `onSongUpdate`

File: app/src/main/java/vn/asiantech/internship/day19/model/Song.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by at-hoavo on 01/07/2017.
+ */
+public class Song implements Parcelable {
+    private String songName;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/model/Song.java]: Please change `songName` to `name`.
Also check ( remove `song` ) on all field on this `model`

File: app/src/main/java/vn/asiantech/internship/day19/model/Utils.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package vn.asiantech.internship.day19.model;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by at-hoavo on 01/07/2017.
+ */
+public class Utils {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/model/Utils.java]: add `final` for this class.
and create a `private constructor` then change `showTime` to `static` method

File: app/src/main/java/vn/asiantech/internship/day19/service/Action.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    private String mValue;
+
+    Action(String value) {
+        mValue = value;
+    }
+
+    public String getValue() {
+        return mValue;
+    }
+}
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/service/Action.java]: add a break line

File: app/src/main/java/vn/asiantech/internship/day19/activity/MusicActivity.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+        for (int i = 0; i < songUrls.length; i++) {
+            mSongs.add(new Song(songNames[i], songArtists[i], songImages[i], songUrls[i], 0));
+        }
+
+        // Create RecyclerView
+        mSongAdapter = new SongAdapter(mSongs, this, this);
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+        mRecyclerView.setAdapter(mSongAdapter);
+        mSongAdapter.notifyDataSetChanged();
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/activity/MusicActivity.java]: remove this line

File: app/src/main/java/vn/asiantech/internship/day19/adapter/SongAdapter.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import vn.asiantech.internship.day19.service.Action;
+import vn.asiantech.internship.day19.service.SongService;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by at-hoavo on 01/07/2017.
+ */
+public class SongAdapter extends RecyclerView.Adapter<SongAdapter.SongViewHolder> {
+    private List<Song> mSongs;
+    private MusicActivity mMusicActivity;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/adapter/SongAdapter.java]: remove this `mMusicActivity`

File: app/src/main/java/vn/asiantech/internship/day19/service/SongService.java [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+
+    private MediaPlayer mMediaPlayer;
+    private List<Song> mSongs;
+    private int mCurrentPosition;
+    private boolean mCheck;
+    private boolean mCheckAutoNext;
+    private boolean mCheckShuffle;
+    private CountDownTimer mCountDownTimer;
+    private NotificationBroadcast mNotificationBroadcast;
+    private Handler mHander = new Handler();
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/service/SongService.java]: `mHander` to `mHandler`

File: app/src/main/java/vn/asiantech/internship/day19/service/SongService.java [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+            public void onTick(long l) {
+                if (mMediaPlayer != null) {
+                    timeIntent.putExtra(MusicActivity.TYPE_TIME, mMediaPlayer.getDuration() + "");
+                    timeIntent.putExtra(MusicActivity.TYPE_SECOND, mMediaPlayer.getCurrentPosition() + "");
+                    timeIntent.putExtra(MusicActivity.TYPE_POSITION, mCurrentPosition);
+                    sendBroadcast(timeIntent);
+                }
+            }
+
+            // No-op
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/service/SongService.java]: please move thís `comment` into method `onFinish`

File: app/src/main/java/vn/asiantech/internship/day19/utils/Utils.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
+package vn.asiantech.internship.day19.utils;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by at-hoavo on 01/07/2017.
+ */
+public final class Utils {
+    private static Utils sUtils;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/utils/Utils.java]: remove  `sUtils`

File: app/src/main/java/vn/asiantech/internship/day19/utils/Utils.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
+package vn.asiantech.internship.day19.utils;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by at-hoavo on 01/07/2017.
+ */
+public final class Utils {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/utils/Utils.java]: class name to `MusicUtil`

**Comment by at-hoavo** [comment]
@at-vinhhuynh please check for me,thank you.

**Comment by at-hoavo** [comment]
@at-vinhhuynh please check for me, thank you!

File: app/src/main/java/vn/asiantech/internship/day19/adapter/OnChooseSongListener.java [hunk 11]
```diff
@@ -0,0 +1,9 @@
+package vn.asiantech.internship.day19.adapter;
+
+/**
+ * Copyright © 2017 AsianTech inc.
+ * Created by rimoka on 03/07/2017.
+ */
+public interface OnChooseSongListener {
+    void OnSongUpdate(int position);
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/day19/adapter/OnChooseSongListener.java]: `OnSongUpdate` to `onSongUpdate`

File: app/src/main/res/layout/activity_music.xml [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+            <ImageButton
+                android:id="@+id/imgBtnPause"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_alignStart="@+id/imgBtnPlay"
+                android:background="@android:color/transparent"
+                android:contentDescription="@null"
+                android:src="@mipmap/img_btn_pause"
+                android:visibility="visible" />
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_music.xml]: remove `android:visibility="visible"`

**Comment by at-hoavo** [comment]
@at-vinhhuynh please check for me,thanks a lot

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I merge this PR, thank you!

**Pull request merged**